### PR TITLE
Update npmignore

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
Updating .npmignore. Adding:
- target
- reports
- tests

We shouldn't need any of the examples/routes from the tests directory for terra-ui.com. The tests directory only contains examples intended for functional verification.
 
https://github.com/cerner/terra-core/issues/384

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-site/.npmignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+node_modules
+src
+target
+reports
+tests
+build
+aggregated-translations
+stats.json
+webpack.config.js
+webpack.prod.config.js
```
Comment (Matt-Butler) [inline packages/terra-site/.npmignore]: Site has a couple additional files that the other packages don't ignore

**Comment by ryanthemanuel** [comment]
Will the test links be broken in the deployed site if we don't include those directories?

**Comment by Matt-Butler** [comment]
> Will the test links be broken in the deployed site if we don't include those directories?

For GitHub pages, we generate the bundle for the deployed site locally. It works on my local machine.

For the review apps, it clones the repo, installs dependencies, and compiles src to lib before generating the bundle. It is working on this PR.

**Comment by bjankord** [comment]
@ryanthemanuel, I can confirm what @Matt-Butler said. I checked out this branch and `npm run clean:install`. Then I ran `npm run deploy` to deploy to gh-pages from this branch. It looks like the deployment successfully kept the test links: http://engineering.cerner.com/terra-core/#/tests/icon-tests/bidi

It also looks like test links are available in the Heroku reviewable apps deployment:
https://terra-core-deployed-pr-415.herokuapp.com/static/#/tests/icon-tests/height-width

**Pull request merged**